### ISSUE TITLE
Indirectly Related Issue: MSD-CDC Composite Bootloader for STM32F103

### ISSUE THREAD

**Issue by SandeepanSengupta** [uTasker/uTasker-Kinetis#5]
Thanks for making it a reality but I still find it a bit less refined and a bit hard to follow. I'd also like to request few features like

1. Secure boot-loader/firmware pair for authentication purposes with STM32 specific tutorial (I couldn't follow the Freescale example tutorial as my experience is limited to Arduino)
2. A cross platform tool (preferably in python) or a header file for ease of compilation
3. If possible, Option to support HID
4. Self-updater feature for bootloader (if possible)
5. A port for STM32F303 (Cortex M4 with FPU in TQFP48)

Although my knowledge is very limited and bounded to Arduino, I'm requesting these features but also I'm willing to help to the best of my ability, in whatever way possible (like testing, making home etched dev board for STM32F303 etc.)
May be I'm asking a lot at a time but to my understanding, these will make a solid platform to work with, far superior to Arduino UNO/Nano based ones we students use in India for school/science-fare projects.

Let me know how may I collaborate

Regards,
Sandeepan

P.S. I have a couple of friends who would love to collaborate too.

**Comment by uTasker** [comment]
Hi

1. The professional version has AES256 protected boot loading - generally the open source one will not have such features since they are reserved for commercial sponsors.
2. There is a header file that is used to configure the project (in fact 2 - one for the features that it should have [config.h] and one for the HW connections and other HW specific details [app_hw_kinetis.h])- A tool is being worked on that will allow these setups to be made graphically and a complete configured project generated; but it doesn't have great priority because it is just as easy to do it from the headers when one has a little experience with the project.
3. HID is already included (see KBOOT USB loader option) - it is shown in a video here: https://www.youtube.com/watch?v=MXsJvTdCcH4&index=25&list=PLWKlVb_MqDQFZAulrUywU30v869JBYi9Q (using its optional AES256 encrypted option)
4. There are various loader options including the "Bare Minimum" loader which may be the feature you are referring to: http://www.utasker.com/docs/uTasker/uTasker_BM_Loader.pdf
It is also possible to update the boot loader from the application (the method to do this is not integrated in the open source version but is available if needed).
5. STM32 support will be added to this repository shortly (allowing projects developed for Kinetis to be run also on STM32 parts - and vice versa)

Note too that the project can be built with Visual Studio (free edition) and the chip is simulated. This allows much faster development, testing and debugging than the traditional IDE tools. It also makes learning/understanding the code and the chip's peripheral operations much easier. Most students and even professionals doen't use these techniques because they haven't yet experienced the huge advantages that it brings - those who do take the time to use it can work much more efficiently than the mainstream coders and therefore become more successful with their work - they just needed to invest a little time to get to know what the others are missing....

Regards

Mark

Mark
